---
- id: snykio:pypi:jinja2:CVE-2014-0012
  title: Privilege Escalation
  description: |
    [`jinja2`][1] is a small but fast and easy to use stand-alone template
    engine written in pure python. FileSystemBytecodeCache in Jinja2 2.7.2
    does not properly create temporary directories, which allows local users
    to gain privileges by pre-creating a temporary directory with a user\'s
    uid.

    **NOTE:** this vulnerability exists because of an incomplete fix for
    [CVE-2014-1402][2].



    [1]: https://pypi.python.org/pypi/jinja2
    [2]: https://snyk.io/vulnSNYK-PYTHON-JINJA2-40028
  affected_package: jinja2
  vulnerable_versions:
  - ">= 2.7.2"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2014-0012
  cwe:
  - CWE-264
  disclosed_date: 2014-01-18
  created_date: 2014-01-18
  last_modified_date: 2014-01-18
  credit:
  - Arun Babu Neelicattu
  references:
  - https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2014-0012
  - https://bugzilla.redhat.com/show_bug.cgi?id=1051421
  - https://github.com/mitsuhiko/jinja2/pull/292
  - https://github.com/mitsuhiko/jinja2/pull/296
  - https://github.com/mitsuhiko/jinja2/commit/acb672b6a179567632e032f547582f30fa2f4aa7
  source_url: https://snyk.io/vuln/SNYK-PYTHON-JINJA2-40250
- id: snykio:pypi:jinja2:CVE-2014-1402
  title: Insecure Defaults
  description: |
    [`Jinja2`][1] is a small but fast and easy to use stand-alone template
    engine written in pure python.

    Affected versions of this package are vulnerable to Insecure Defaults
    attacks. The default configuration for bccache.FileSystemBytecodeCache
    in Jinja2 before 2.7.2 does not properly create temporary files, which
    allows local users to gain privileges via a crafted .cache file with a
    name starting with `jinja2` in /tmp.



    [1]: https://pypi.python.org/pypi/Jinja2
  affected_package: jinja2
  vulnerable_versions:
  - "< 2.7.2"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2014-1402
  cwe:
  - CWE-264
  disclosed_date: 2014-01-10
  created_date: 2014-01-10
  last_modified_date: 2014-01-10
  credit:
  - Unknown
  references:
  - https://bugzilla.redhat.com/CVE-2014-1402
  - http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=734747
  - https://github.com/mitsuhiko/jinja2/commit/acb672b6
  source_url: https://snyk.io/vuln/SNYK-PYTHON-JINJA2-40028
